src/Propulsion.SqlStreamStore/StreamReader.fs

```diff
@@ -0,0 +1,226 @@
+namespace Propulsion.SqlStreamStore
+
+open System
+open System.Text
+
+open SqlStreamStore
+open SqlStreamStore.Streams
+
+open Serilog
+
+open System.Diagnostics
+open Propulsion.Streams
+
+[<AutoOpen>]
+module private Internal =
+
+    [<NoComparison;NoEquality>]
+    type InternalBatch =
+        {
+            firstPosition  : int64
+            lastPosition   : int64
+            messages       : StreamEvent<byte []> array
+            isEnd          : bool
+        }
+        member this.Length = this.messages.Length
+
+    [<RequireQualifiedAccess>]
+    module StreamMessage =
+
+        let intoStreamEvent (msg: StreamMessage) : Propulsion.Streams.StreamEvent<_> =
+            let inline len0ToNull (x : _[]) =
+                match x with
+                | null -> null
+                | x when x.Length = 0 -> null
+                | x -> x
+
+            let data =
+                let json = msg.GetJsonData().Result
+                Encoding.UTF8.GetBytes(json)
+
+            let metadata =
+                Encoding.UTF8.GetBytes(msg.JsonMetadata)
+
+            let event =
+                FsCodec.Core.TimelineEvent.Create (
+                    msg.Position,
+                    msg.Type,
+                    len0ToNull data,
+                    len0ToNull metadata,
+                    msg.MessageId,
+                    timestamp = DateTimeOffset(msg.CreatedUtc)
+                )
+
+            { stream = StreamName.internalParseSafe msg.StreamId; event = event }
+
+    type Stats(logger: ILogger, ?statsInterval: TimeSpan) =
```

Comment:
newer code pushes this defaulting outward (i.e. it goes into a `type Arguments` in `dotnet-templates` )

Reply:
so what specifically do you prefer here? Make statsInterval non-optional?

Reply:
Yes, non-optional (I pushed a lot of these out about a month ago as there are better ways to make the consumer logic succinct than having defaults inside Propulsion itself)